### ISSUE TITLE
Reconnect spam when socket closes and fails to reconnect

### ISSUE THREAD

**Issue by deluvas** [tompetko/SignalR.WebSocketSharpTransport#3]
When the connection is lost, and the transport tries to reconnect but fails, it doesn't wait the reconnect delay and tries to immediately connect again, which results in spam.

The flow:
````
connection lost -> websocket.OnClose -> DoReconnect -> websocket.ConnectAsync -> failure to connect -> repeat
````

The expected behavior is for the transport to delay the next reconnect.

**Comment by deluvas** [comment]
As a quick fix, I added the following code to check the time before reconnecting. I don't know how well it works, but it's well enough for my needs.

````C#
// after while (TransportHelper.VerifyLastActive(_connection) && _connection.EnsureReconnecting()) 
if ( _lastReconnectTime + ReconnectDelay >= DateTime.Now ) {
    await Task.Delay( DateTime.Now - _lastReconnectTime + ReconnectDelay );
}
````
and 
````
_lastReconnectTime = DateTime.Now; // in PerformConnect
````

But my understanding is limited and maybe it's wrong, so perhaps you guys can find a better solution.

**Comment by tompetko** [comment]
Fixed by commit a5ddcba0e4dca7b9ab648cb07da8a8a28c59df0c. If problem persists, reopen this issue with detailed steps to reproduce.
